recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,171 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration, ConanException
+
+class SociConan(ConanFile):
+    name = "soci"
+    homepage = "https://github.com/SOCI/soci"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "The C++ Database Access Library "
+    topics = ("cpp", "database-library", "oracle", "postgresql", "mysql", "odbc", "db2", "firebird", "sqlite3", "boost" )
+    license = "BSL-1.0"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    options = {
+        "fPIC":             [True, False],
+        "shared":           [True, False],
+        "empty":            [True, False],
+        "with_sqlite3":     [True, False],
+        "with_db2":         [True, False],
+        "with_odbc":        [True, False],
+        "with_oracle":      [True, False],
+        "with_firebird":    [True, False],
+        "with_mysql":       [True, False],
+        "with_postgresql":  [True, False],
+        "with_boost":       [True, False]
+    }
+
+    default_options = {
+        "fPIC":             True,
+        "shared":           False,
+        "empty":            False,
+        "with_sqlite3":     False,
+        "with_db2":         False,
+        "with_odbc":        False,
+        "with_oracle":      False,
+        "with_firebird":    False,
+        "with_mysql":       False,
+        "with_postgresql":  False,
+        "with_boost":       False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "19",
+            "gcc": "5.0",
+            "clang": "3.8",
+            "apple-clang": "10.0",
```

Comment:
```suggestion
            "Visual Studio": "14",
            "gcc": "4.8",
            "clang": "3.8",
            "apple-clang": "8.0",
```
https://en.cppreference.com/w/cpp/compiler_support